<?xml version="1.0"?>
<robot name="robot">

    <material name="white">
        <color rgba="1 1 1 1" />
    </material>

    <material name="blue">
        <color rgba="0.2 0.2 1 1"/>
    </material>

    <material name="purple">
        <color rgba="0.639 0.043 0.639 1"/>
    </material>


    <link name="base_link" />

    <joint name="arm1_joint" type="continuous">
        <parent link="base_link"/>
        <child link="arm1"/>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <axis xyz="0 0 1"/>
    </joint>

    <link name="arm1">
        <visual>
            <origin xyz="0.25 0 0" rpy="0 0 0" />
            <geometry>
                <box size="0.5 0.01 0.01"/>
            </geometry>
            <material name="blue"/>
        </visual>
        <collision>
            <origin xyz="0.25 0 0" rpy="0 0 0" />
            <geometry>
                <box size="0.5 0.01 0.01"/>
            </geometry>
        </collision>
    </link>

    <joint name="arm2_joint" type="continuous">
        <parent link="arm1"/>
        <child link="arm2"/>
        <origin xyz="0.5 0 0" rpy="0 0 0" />
        <axis xyz="0 0 1"/>
    </joint>

     <link name="arm2">
        <visual>
            <origin xyz="0.25 0 0" rpy="0 0 0" />
            <geometry>
                <box size="0.5 0.01 0.01"/>
            </geometry>
            <material name="purple"/>
        </visual>
        <collision>
            <origin xyz="0.25 0 0" rpy="0 0 0" />
            <geometry>
                <box size="0.5 0.01 0.01"/>
            </geometry>
        </collision>
    </link>

    <link name="ef"/>
    <joint name="ef_joint" type="continuous">
        <parent link="arm2"/>
        <child link="ef"/>
        <origin xyz="0.5 0 0" rpy="0 0 0" />
        <axis xyz="0 0 1"/>
    </joint>

</robot>